contracts/extensions/OneTxPayment.sol

```diff
@@ -18,34 +18,31 @@
 pragma solidity 0.5.8;
 pragma experimental ABIEncoderV2;
 
-import "./../ColonyAuthority.sol";
 import "./../ColonyDataTypes.sol";
 import "./../IColony.sol";
-import "./../IColonyNetwork.sol";
 
 // ignore-file-swc-108
 
 contract OneTxPayment {
-  bytes4 constant ADD_PAYMENT_SIG = bytes4(keccak256("addPayment(uint256,uint256,address,address,uint256,uint256,uint256)"));
```

Comment:
Looking back it was hasty to knee-jerk think that expenditures would help reduce LeapDAOs gas, since they were using the payments-based OneTx, not the tasks-based OneTx (which we never deployed). I'm not sure if we can get the gas costs down much lower then they are with Payments/Expenditures currently... let me think about it.

Reply:
I've taken a stab at reimplementing their contract so it uses expenditures rather than the `OneTxPayment.sol`, and my first stab goes from `1647114` gas to `708741` gas without even really thinking about it so definitely worth them switching 😄

Reply:
Wow! Can I see the change?

One (tiny) change I was able to make to `OneTxPayment` was to fund the task _before_ setting the payout, which avoids incrementing/decrementing the `payouts we cannot make` counter.